modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/SubscriptionsFilterByStatus/index.js

```diff
@@ -1,61 +1,89 @@
 import ClayButton from '@clayui/button';
-import {ClayDropDownWithItems} from '@clayui/drop-down';
-import React, {useEffect} from 'react';
-
-const SubscriptionsFilterByStatus = ({setSelectedStatus}) => {
-	const items = [
-		{
-			items: [
-				{
-					checked: true,
-					label: 'All',
-					onChange: () => alert('All'),
-					type: 'checkbox',
-				},
-				{
-					checked: false,
-					label: 'Active',
-					onChange: () => alert('Active'),
-					type: 'checkbox',
-				},
-				{
-					checked: false,
-					label: 'Future',
-					onChange: () => alert('Future'),
-					type: 'checkbox',
-				},
-				{
-					checked: false,
-					label: 'Expired',
-					onChange: () => alert('Expired'),
-					type: 'checkbox',
-				},
-			],
-			type: 'group',
-		},
-	];
-
-	// const handleFilterChange = (event) => {
-	// 	setSelectedStatus(event.target.value);
-	// };
-
-	useEffect(() => {
-		setSelectedStatus('All');
-	}, [setSelectedStatus]);
+import {DropDown} from '@clayui/core';
+import React, {useState} from 'react';
+import {POSSIBLE_STATUS_AMOUNT} from '../Subscriptions';
+
+const SubscriptionsFilterByStatus = ({selectedStatus, setSelectedStatus}) => {
+	const [active, setActive] = useState(false);
+
+	const handleChange = (status) => {
+		if (status === 'All') {
+			setSelectedStatus(
+				selectedStatus.length === POSSIBLE_STATUS_AMOUNT
+					? []
+					: ['Active', 'Expired', 'Future']
+			);
+		} else {
```

Comment:
```suggestion
		}
```

Else not needed, since we can return above